### PR TITLE
Get previous email address(es) from SSO accounts request

### PR DESCRIPTION
**Pull request by MiguelAHM**
We need old emails for sending them in the surf requests.
For that we need to modify the call to get those values as well

### CODE REVIEW

**Review by eshryane — APPROVED** [review]

File: whois-api/src/test/java/net/ripe/db/whois/api/sso/AuthServiceClientTestIntegration.java [hunk 1]
```diff
@@ -17,6 +18,7 @@
 @Tag("IntegrationTest")
 public class AuthServiceClientTestIntegration extends AbstractIntegrationTest {
 
+    private final static String UUID = "8ffe29be-89ef-41c8-ba7f-0e1553a623e5";
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/sso/AuthServiceClientTestIntegration.java:21]: Use a canonical order for modifiers: `private static final`

Ref. https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Method.html

File: whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java [hunk 2]
```diff
@@ -162,6 +165,32 @@ public ValidateTokenResponse getUserDetails(final String uuid) {
         }
     }
 
+    @Cacheable(cacheNames="ssoHistoricalUserDetails", key="#uuid")
+    public HistoricalUserResponse getHistoricalUserDetails(final String uuid) {
+        if (StringUtils.isEmpty(uuid)) {
+            LOGGER.debug("No uuid was supplied");
+            throw new AuthServiceClientException(BAD_REQUEST.getStatusCode(),"Invalid uuid.");
```
Comment (eshryane) [inline whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java:172]: Return "No UUID" rather than "Invalid UUID" ?

**Review by eshryane — APPROVED** [review]

**Comment by MiguelAHM** [comment]
test ok
https://gitlab.ripe.net/swe-database-team/whois/-/pipelines/483839

File: whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java [hunk 3]
```diff
@@ -162,6 +165,32 @@ public ValidateTokenResponse getUserDetails(final String uuid) {
...
+        }
+
+        try {
+            return client.target(restUrl)
+                    .path(HISTORICAL_USER_SEARCH_PATH)
+                    .path(uuid)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .header("X-API_KEY", apiKey)
```
Comment (eshryane) [inline whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java:180]: Is the query parameter `format=json` not necessary?

Comment (MiguelAHM) [reply whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java:196]: no, it is not required (tested with insomnia)

File: whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java [hunk 4]
```diff
@@ -162,6 +165,32 @@ public ValidateTokenResponse getUserDetails(final String uuid) {
...
+                    .get(HistoricalUserResponse.class);
+        } catch (BadRequestException e) {
+            LOGGER.debug("Failed to get details for uuid {} (token is invalid)", uuid);
```
[the remark this answers is not in the document]
Comment (MiguelAHM) [reply whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java]: Done

**Review by eshryane — APPROVED** [review]

**Pull request merged**